game_of_life/game.py

```diff
@@ -0,0 +1,176 @@
+import pygame
+
+# Create own type of matrix
+Matrix = list[list[int]]
+
+
+class Game:
+    '''Class to create the game of life'''
+
+    def __init__(self,
+                 state: Matrix,
+                 display_size_x: int,
+                 display_size_y: int,
+                 death_cell_color: tuple,
+                 alive_cell_color: tuple,
+                 fps: int):
+        '''constructor to initialize data'''
+
+        # Initialize the first frame matrix
+        self._state: Matrix = state
+
+        # Initialize the framerate
+        self._fps: int = fps
+
+        # Define colors
+        self._death_cell_color: tuple = death_cell_color
+        self._alive_cell_color: tuple = alive_cell_color
+
+        # Initialize the game window size
+        self._display_size_x: int = display_size_x
+        self._display_size_y: int = display_size_y
+
+        # Initialize the cells size
+        self._cell_size_x: int = int(display_size_x / len(self._state[0]))
+        self._cell_size_y: int = int(display_size_y / len(self._state))
+
+        # Setup pygame stuff
+        self._clock: pygame.time.Clock = pygame.time.Clock()
+        self._done: bool = True
+        self._screen: pygame.surface.Surface
+
+    def start(self):
+        '''Public function to start the game'''
+
+        print("The game starting.")
+
+        self._done = False
+
+        # Create the game window
+        self._screen = pygame.display.set_mode(
+            (self._display_size_x, self._display_size_y)
+            )
+        self._screen.fill(self._death_cell_color)
+
+        pygame.init()
+        self._update()
+
+    def _update(self):
+        '''Private function which display game for each frame '''
+
+        # While the game is not over
+        while not self._done:
+
+            # Listen for all events
+            for event in pygame.event.get():
+
+                # Quit the infinite loop when the user presses the close button
+                if event.type == pygame.QUIT:
+                    self._done = True
+
+            # For each iteration
+            for index_line in range(len(self._state)):
+                for index_column in range(len(self._state[index_line])):
+                    if self._state[index_line][index_column] == 1:
```

Comment:
on entre dans un if dans un for dans un for dans un white dans un def : il faut des sous fonctions ;)